### PR TITLE
SECURITY: Limit user profile field length

### PR DESCRIPTION
**Pull request by martin-brennan**
Adds limits to location and website fields at model and DB level to match the bio_raw field limits. A limit cannot be added at the DB level for bio_raw because it is a postgres text field.

### CODE REVIEW

File: db/migrate/20220920044310_enforce_user_profile_max_limits.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class EnforceUserProfileMaxLimits < ActiveRecord::Migration[7.0]
+  def change
+    execute "UPDATE user_profiles SET location = LEFT(location, 3000)"
```
Comment (tgxworld) [inline db/migrate/20220920044310_enforce_user_profile_max_limits.rb:5]:
```suggestion
    execute "UPDATE user_profiles SET location = LEFT(location, 3000) WHERE location IS NOT NULL"
```

File: db/migrate/20220920044310_enforce_user_profile_max_limits.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    execute "UPDATE user_profiles SET website = LEFT(website, 3000)"
```
Comment (tgxworld) [inline db/migrate/20220920044310_enforce_user_profile_max_limits.rb:6]:
```suggestion
    execute "UPDATE user_profiles SET website = LEFT(website, 3000) WHERE website IS NOT NULL"
```

File: spec/models/user_profile_spec.rb [hunk 3]
```diff
@@ -42,6 +42,15 @@
         end
       end
 
+      context "when it is > 3000 characters" do
+        before { profile.location = "a" * 3500 }
+
+        it "is not valid" do
+          profile.valid?
```
Comment (tgxworld) [inline spec/models/user_profile_spec.rb:49]:
```suggestion
          expect(profile.valid?).to eq(false)
```

File: spec/models/user_profile_spec.rb [hunk 4]
```diff
@@ -63,6 +72,15 @@
         end
       end
 
+      context "when it is > 3000 characters" do
+        before { profile.bio_raw = "a" * 3500 }
+
+        it "is not valid" do
+          profile.valid?
```
Comment (tgxworld) [inline spec/models/user_profile_spec.rb:79]:
```suggestion
          expect(profile.valid?).to eq(false)
```

**Comment by martin-brennan** [comment]
@tgxworld agree with the reviews, made a commit to address :+1:

**Review by tgxworld — APPROVED** [review]

**Pull request merged**